python/orca/tutorial/NCF/utils.py

```diff
@@ -0,0 +1,93 @@
+#
+# Copyright 2016 The BigDL Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import argparse
+
+from bigdl.orca import init_orca_context
+
+
+def parse_args(description):
+    parser = argparse.ArgumentParser(description=description)
+    parser.add_argument("--data_dir", type=str,
+                        help="The path to load data from local or remote resources.")
+    parser.add_argument("--cluster_mode", type=str, default="local",
+                        help="The cluster mode, such as local, yarn-client, yarn-cluster, "
+                             "k8s-client, k8s-cluster, spark-submit or bigdl-submit.")
+    parser.add_argument("--backend", type=str, default="spark", help="ray or spark")
+    parser.add_argument("--tensorboard", action='store_true',
+                        help="Whether to use TensorBoardCallback.")
+    parser.add_argument("--workers_per_node", type=int, default=1,
+                        help="The number of workers on each node.")
+    args = parser.parse_args()
+    return args
+
+
+def init_orca(args, extra_python_lib=None):
+    if args.cluster_mode == "local":
+        sc = init_orca_context(cluster_mode="local")
+    elif args.cluster_mode.startswith("yarn"):
+        if args.cluster_mode == "yarn-client":
+            sc = init_orca_context(cluster_mode="yarn-client",
+                                   cores=4, memory="10g", num_nodes=2,

```

Comment:
is 10g necessary? can 2g or 4g work?

Reply:
xshards needs at least 10g on yarn clusters with cluster_mode="local", "yarn-client" or "yarn-cluster".

Reply:
okay, will investigate it later